cg/clients/freshdesk/freshdesk_client.py

```diff
@@ -11,28 +19,47 @@
 class FreshdeskClient:
     """Client for communicating with the freshdesk REST API."""
 
-    def __init__(self):
-        self.headers = None
-        self.api_key = None
-        self.url = None
-
-    def init_app(self, url: str, api_key: str):
-        """Set up the client."""
-        self.url = url
+    def __init__(self, base_url: str, api_key: str):
+        self.base_url = base_url
         self.api_key = api_key
-        self.headers = {"Content-Type": "application/json"}
-
-    @property
-    def auth_header(self):
-        return self.api_key, "X"
 
     def create_ticket(self, ticket: TicketCreate) -> TicketResponse:
         """Create a ticket."""
         LOG.debug(ticket.model_dump_json())
-        response = requests.post(
-            url=f"{self.url}{EndPoints.TICKETS}",
-            headers=self.headers,
-            auth=self.auth_header,
-            json=ticket.model_dump(exclude_none=True),
+        session: Session = self._get_session()
```

Comment:
I'd instantiate the session in the constructor and keep it there to get more value from the pooled TCP connections.

Reply:
Done 👍